### PR TITLE
Do not use VLAs in `cp_async_bulk_tensor_*` tests

### PR DESCRIPTION
**Pull request by miscco**
VLAs are a compiler extension and are correctly errored out by some compilers. As we always know the exact size of the array anyway just swtich to a `cuda::std::array`

### CODE REVIEW

**Review by miscco — COMMENTED** [review]
I obviously also managed to smuggle in a type int the `__cccl_ptx_isa` macro, this also fixes that

File: libcudacxx/include/cuda/std/detail/libcxx/include/__cccl/ptx_isa.h [hunk 1]
```diff
@@ -98,7 +98,7 @@
 // When __CUDA_MINIMUM_ARCH__ is available, we only enable the feature when the
 // hardware supports it.
 #if (!defined(__CUDA_MINIMUM_ARCH__)) \
-  || (defined(__CUDA_MINIMUM_ARCH__) && 900 <= __CUDA_MINIMUM_ARCH__) && __cccl_isa_ptx >= 800
+  || (defined(__CUDA_MINIMUM_ARCH__) && 900 <= __CUDA_MINIMUM_ARCH__) && __cccl_ptx_isa >= 800
```
Comment (miscco) [inline libcudacxx/include/cuda/std/detail/libcxx/include/__cccl/ptx_isa.h:101]: 🤦‍♂️

File: libcudacxx/test/libcudacxx/cuda/barrier/cp_async_bulk_tensor_1d.pass.cpp [hunk 2]
```diff
@@ -54,9 +56,10 @@ int main(int, char**)
         ),
         NV_IS_DEVICE, (
             for (auto smem_coord : TEST_SMEM_COORDS) {
-                test<smem_len>(smem_coord, SMEM_DIMS_DEV, GMEM_DIMS_DEV, gmem_tensor, gmem_len);
+                test(smem_coord, SMEM_DIMS_DEV, GMEM_DIMS_DEV, gmem_tensor, gmem_len);
             }
         )
     );
+    (void)SMEM_DIMS;
```
Comment (miscco) [inline libcudacxx/test/libcudacxx/cuda/barrier/cp_async_bulk_tensor_1d.pass.cpp:63]: We do not use SMEM_DIMS on the device path so it is getting flagged as unused

File: libcudacxx/test/libcudacxx/cuda/barrier/cp_async_bulk_tensor_generic.h [hunk 3]
```diff
@@ -129,10 +130,10 @@ __constant__ fake_cutensormap global_fake_tensor_map;
  * 5. It writes the tile back to global memory
  * 6. It checks that all the values in global are properly modified.
  */
-template <int smem_len>
-__device__ void test(std::initializer_list<int> smem_coord,
-                     std::initializer_list<int> smem_dims,
-                     std::initializer_list<int> gmem_dims,
+template <size_t smem_len, size_t num_dims>
+__device__ void test(cuda::std::array<uint32_t, smem_len> smem_coord,
```
Comment (ahendriksen) [inline libcudacxx/test/libcudacxx/cuda/barrier/cp_async_bulk_tensor_generic.h:134]:
```suggestion
__device__ void test(cuda::std::array<uint32_t, num_dims> smem_coord,
```

smem_len is the size of the shared memory in bytes. It should not be set by the number of dimensions of the shared memory coordinate.

**Review by ahendriksen — APPROVED** [review]
LGTM!

File: libcudacxx/test/libcudacxx/cuda/barrier/cp_async_bulk_tensor_1d.pass.cpp (hunk 2 above)
Comment (miscco) [reply libcudacxx/test/libcudacxx/cuda/barrier/cp_async_bulk_tensor_1d.pass.cpp]: dropped that part because we actually needed it ^^

**Review by wmaxey — APPROVED** [review]

**Pull request merged**